daq/stream_monitor.py

```diff
@@ -145,16 +149,18 @@ def event_loop(self):
                 LOGGER.error('Monitoring exception in callback: %s', e)
                 LOGGER.exception(e)
             self.log_monitors()
-            self.dirty = False
-            fds = self.poller.poll(self.timeout_ms)
+            fds = self.poller.poll(self.timeout_sec * 10e3 if self.timeout_sec else None)
             LOGGER.debug('Monitoring found fds %s', fds)
             if fds:
                 for fd, event in fds:
-                    self.process_poll_result(event, fd)
-                    if self.dirty:
-                        LOGGER.debug('Monitoring set modified, re-issuing poll...')
-                        break
-            else:
-                LOGGER.debug('Monitoring no fds found')
-                return True
+                    if fd in self.callbacks:
+                        self.process_poll_result(event, fd) #Monitoring set could be modified
```

Comment:
space after #, also move to previous line to indicate *why* the callback list is checked.  (e.g., could have been modified so the callbacks are no longer valid).